bioconda_utils/lint/checks.py

```diff
@@ -0,0 +1,552 @@
+    """The extra/skip-lints section must contain a list"""
+    def check_recipe(self, recipe):
+        if not isinstance(recipe.get('extra/skip-lints', []), list):
+            self.message(section='extra/skip-lints')
+
+
+class deprecated_numpy_spec(LintCheck):
+    """The recipe contains a deprecated numpy spec
+
+    Please remove the ``x.x`` - pinning is now handled automatically.
+
+    """
+    def check_deps(self, deps):
+        if 'numpy' not in deps:
+            return
+        for path in deps['numpy']:
+            line, _, _ = self.recipe.get_raw(path).partition('#')
+            if 'x.x' in line:
+                self.message(section=path)
+
+
+class should_not_use_fn(LintCheck):
+    """The recipe uses source/fn
+
+    There is no need to specify the filename as the URL should give a name
+    and it will in most cases be unpacked automatically.
+    """
+    def check_source(self, source, section):
+        if 'fn' in source:
+            self.message(section=section+'/fn')
+
+
+class should_use_compilers(LintCheck):
+    """The recipe requires a compiler directly
+
+    Since version 3, ``conda-build`` uses a special syntax to require
+    compilers for a given language matching the architecture for
+    which a package is being build. Please use::
+
+        requirements:
+           build:
+             - {{ compiler('language') }}
+
+    Where language is one of ``c``, ``cxx``, ``fortran``, ``go`` or
+    ``cgo``. You can specify multiple compilers if needed.
+
+    There is no need to add ``libgfortran``, ``libgcc``, or
+    ``toolchain`` to the dependencies as this will be handled by
+    conda-build itself.
+
+    """
+    compilers = ('gcc', 'llvm', 'libgfortran', 'libgcc', 'go', 'cgo',
+                 'toolchain')
+
+    def check_deps(self, deps):
+        for compiler in self.compilers:
+            for location in deps.get(compiler, []):
+                self.message(section=location)
+
+
+class compilers_must_be_in_build(LintCheck):
+    """The recipe requests a compiler in a section other than build
+
+    Please move the ``{{ compiler('language') }}`` line into the
+    ``requirements: build:`` section.
+
+    """
+    def check_deps(self, deps):
+        for dep in deps:
+            if dep.startswith('compiler_'):
+                for location in deps[dep]:
+                    if 'run' in location or 'host' in location:
+                        self.message(section=location)
+
+
+class recipe_is_blacklisted(LintCheck):
+    """The recipe is currently blacklisted and will not be built.
+
+    If you are intending to repair this recipe, remove it from
+    the build fail blacklist.
+    """
+    def __init__(self, linter):
+        super().__init__(linter)
+        self.blacklist = linter.get_blacklist()
+
+    def check_recipe(self, recipe):
+        if recipe.name in self.blacklist:
+            self.message(section='package/name')
+
+
+class folder_and_package_name_must_match(LintCheck):
```

Comment:
Is this a new one? Good idea.

Reply:
Thanks, yes. Had a ticket, and it was easy to do now.